UiPath.FreeRdpClient/UiPath.FreeRdpClient/Internals/FreeRdpClient.cs

```diff
@@ -1,20 +1,47 @@
-﻿using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.Hosting;
-using Microsoft.Extensions.Logging;
+﻿using Microsoft.Extensions.Logging;
+using Nito.AsyncEx;
 using Nito.Disposables;
 using System.Net.Sockets;
 
 namespace UiPath.Rdp;
 
-public static class FreeRdpClient
+internal class FreeRdpClient : IFreeRdpClient
 {
     static FreeRdpClient()
     {
         //Make sure winsock is initialized
         using var _ = new TcpClient();
     }
 
-    public static async Task<IAsyncDisposable> Connect(RdpConnectionSettings connectionSettings)
+    public FreeRdpClient(ILogger<FreeRdpClient> logger)
+    {
+        _log = logger;
+    }
+
+    private readonly ILogger<FreeRdpClient> _log;
+    private static AsyncLock? InitLock = new();
+
+    private async Task<IDisposable?> TryAcquireInitLock()
+    {
+        /// Make sure freerdp static initilizers are not run concurrently
+        /// when they do they fail with 
+        if (InitLock is null)
+            return null;
+
+        var releaseLock = await InitLock.LockAsync();
+        if (InitLock is null)
```

Comment:
2/🧵 I don't see the reason behind this mechanism, and I think we can fix whatever the problem differently.

⚠️ Should we not find a better solution:
Even though references are the same size as IntPtr which means that assigning is atomic, this doesn't fix the multicore with distinct caches problem, and we should still use synchronization.